components/payments.tsx

```diff
@@ -139,31 +127,20 @@ function getLineItems(invoiceOrSession: Stripe_Invoice | Stripe_CheckoutSession)
 }
 
 export const PaymentItemCard = ({
-  payment: { created, invoice, currency, amount, session, shipment }
+  payment: { created, invoiceItems, currency, amount, sessionItems, shipment }
 }: {
   payment: Stripe_PaymentIntent;
 }) => {
-  invoice = invoice as Stripe_Invoice;
   // Only subscriptions will have `session`, and one-off purchases will only have `session`.
   // We get the same data from both, so we collapse here, preferring the `invoice`.
-  const lineItems = getLineItems(invoice ?? session);
+  const lineItems = getLineItems(invoiceItems.length ? invoiceItems : sessionItems);
 
   return (
     <Card sx={{ width: '100%' }}>
       <Flex sx={{ alignItems: 'center', justifyContent: 'space-between' }}>
         <Paragraph variant="smallHeading" sx={{ lineHeight: '2' }}>
           {format(created * 1000, 'PP')}
         </Paragraph>
-        <Box sx={{ textAlign: 'right' }}>
-          {invoice?.invoice_pdf ? (
-            <Link title="Download Receipt" target="_blank" rel="noreferrer" href={invoice?.invoice_pdf}>
-              <IconButton color="text">
-                <BsReceipt size={16} />
-              </IconButton>
-            </Link>
-          ) : null}
-          {invoice?.paid ? <BsCheckCircleFill title="Paid" color="green" size={16} /> : null}
-        </Box>
```

Comment:
Losing the link to the invoice PDF feels like a downgrade. Any way to keep this functionality?

Reply:
The only way I could get to a reasonable depth was to just get invoice items and exclude the invoice. The indexing set up is pretty fragile right now.

We only have invoices for a subset of purchases  — subscriptions — so it seemed like an acceptable loss.